### PR TITLE
Renames `set_events` to `add_events_from`

### LINKED ISSUE #4266: Rename `set_events`
As discussed in the PR #4254, `set_event` should be renamed because of misleading.

              This adds a side effect to what would be expected to be a simple setter. At minimum it should be changed to `append_events()` or `add_events()`. Was there an actual bug that this was fixing or just not sure if it would cause issues?

_Originally posted by @kratman in https://github.com/pybamm-team/PyBaMM/pull/4254#discussion_r1672332716_

### PR DESCRIPTION
**Pull request by medha-14**
# Description

I have renamed `set_events` to `add_events_from`  in all occurrences

Fixes #4266 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All relevant tests pass, except latxeify which are unrelated.
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

### CODE REVIEW

**Review by kratman — CHANGES_REQUESTED** [review]
It looks like you might have missed `reaction_driven_porosity.py` when doing these changes. Can you double check that one?

**Comment by medha-14** [comment]
> It looks like you might have missed `reaction_driven_porosity.py` when doing these changes. Can you double check that one?

Sorry I missed it.

**Comment by kratman** [comment]
Thank you, just update the changelog and it should be good to go. Put this one under "breaking changes" in the unreleased section since it is renaming a function.

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@
   `have_iree` with `has_iree` ([#4398](https://github.com/pybamm-team/PyBaMM/pull/4398))
 - Remove deprecated function `pybamm_install_jax` ([#4362](https://github.com/pybamm-team/PyBaMM/pull/4362))
 - Removed legacy python-IDAKLU solver. ([#4326](https://github.com/pybamm-team/PyBaMM/pull/4326))
+- Renamed `set_events` function to `add_events_from` to better reflect its purpose. This may require updating calls to set_events in your codebase. ([#4421](https://github.com/pybamm-team/PyBaMM/pull/4421))
```
Comment (kratman) [inline CHANGELOG.md]: This is in the wrong section. It should be under "unreleased" at the top. You will just need to create a new breaking changes section for it.

This part is for stuff in the last release

Comment (medha-14) [reply CHANGELOG.md]: Yes, I understand. Thank you!

**Review by kratman — APPROVED** [review]
Thanks looks good

**Comment by agriyakhetarpal** [comment]
@all-contributors please add @medha-14 for code

**Comment by medha-14** [comment]
Thank you, @kratman, for the review. This was my first PR, and I look forward to contributing more to Pybamm.

**Pull request merged**